### PR TITLE
Add FirstPopulated property

### LINKED ISSUE #62: ASCollection: add property to get first-populated of (Current, First, Last)
This is a gap in our API / null safety. If IsPaged is true, then *at least* one of these must be true. We can define that meta property and link its nullability to IsPaged. This will provide a slightly more ergonomic API over paged collections.

### PR DESCRIPTION
**Pull request by CreggHancock**
Resolves #62 

As of now I'm just adding the property to both classes and adding the `MemberNotNullWhen` attribute to `IsPaged`. I'm not sure if I'm overthinking it or something, but I feel like there's something more desired here that I'm missing. So feedback would be appreciated!

### CODE REVIEW

File: Source/ActivityPub.Types/AS/Collection/ASCollection.cs [hunk 1]
```diff
@@ -112,8 +112,14 @@ public int TotalItems
     /// <summary>
     ///     True if this is a paged collection, false otherwise.
     /// </summary>
+    [MemberNotNullWhen(true, nameof(FirstPopulated))]
     public bool IsPaged => Current != null || First != null || Last != null;
 
+    /// <summary>
+    ///     In a paged Collection, gets the first populated entity.
+    /// </summary>
+    public Linkable<ASCollectionPage>? FirstPopulated => IsPaged ? Current ?? First ?? Last : null;
```
Comment (warriordog) [inline Source/ActivityPub.Types/AS/Collection/ASCollection.cs:121]: We should be able to remove the IsPaged check and simplify it down to just `Current ?? First ?? Last`

File: Source/ActivityPub.Types/AS/Collection/ASOrderedCollection.cs [hunk 2]
```diff
@@ -113,8 +113,14 @@ public int TotalItems
     /// <summary>
     ///     True if this is a paged collection, false otherwise.
     /// </summary>
+    [MemberNotNullWhen(true, nameof(FirstPopulated))]
     public bool IsPaged => Current != null || First != null || Last != null;
 
+    /// <summary>
+    ///     In a paged Collection, gets the first populated entity.
+    /// </summary>
+    public Linkable<ASOrderedCollectionPage>? FirstPopulated => IsPaged ? Current ?? First ?? Last : null;
```
Comment (warriordog) [inline Source/ActivityPub.Types/AS/Collection/ASOrderedCollection.cs:122]: same thing

**Review by warriordog — CHANGES_REQUESTED** [review]
thank you for the PR! I left a comment, but it's a small thing.

File: Source/ActivityPub.Types/AS/Collection/ASOrderedCollection.cs (hunk 2 above)
Comment (CreggHancock) [reply Source/ActivityPub.Types/AS/Collection/ASOrderedCollection.cs]: I guess I just had the IsPaged check there for readability, but totally makes sense to remove it. Just pushed up these fixes.

**Review by warriordog — APPROVED** [review]
@CreggHancock looks good, approved. Thank you for the PR!

**Pull request merged**